### PR TITLE
Add rows background Color

### PR DESCRIPTION
**Pull request by titanve**
This PR is for putting `backgroundColor` for the Calendar Rows.

### Screenshot of what is intended with this PR:
image

### CODE REVIEW

**Comment by titanve** [comment]
@acro5piano I'm not sure if it should be done in the component `_HourGuideColumn`, can you confirm that I'm in the right direction? 🙏

File: src/components/HourGuideColumn.tsx [hunk 1]
```diff
@@ -9,17 +9,21 @@ interface HourGuideColumnProps {
...
 }
 
-const _HourGuideColumn = ({ cellHeight, hour, ampm }: HourGuideColumnProps) => {
+const _HourGuideColumn = ({ cellHeight, hour, ampm, index }: HourGuideColumnProps) => {
   const theme = useTheme()
   const textStyle = React.useMemo(
     () => ({ color: theme.palette.gray[500], fontSize: theme.typography.xs.fontSize }),
     [theme],
   )
+  const is_pair = (i) => i % 2
```
Comment (acro5piano) [inline src/components/HourGuideColumn.tsx:21]: - is_pair should be `isPair`
- This function should be outside this component
- `(i: number)` to not be implicit `any`

File: src/theme/ThemeInterface.ts [hunk 2]
```diff
@@ -8,6 +8,8 @@ import { DeepPartial } from '../utility-types'
 export interface Palette {
   main: string
   contrastText: string
+  pairCellBg: string
```
Comment (acro5piano) [inline src/theme/ThemeInterface.ts:11]: `pairCellBg` should be `evenCellBg`, right? 
I'm not a native English speaker so I may be wrong!

**Review by acro5piano — CHANGES_REQUESTED** [review]
Thanks!

File: src/theme/ThemeInterface.ts (hunk 2 above)
Comment (titanve) [reply src/theme/ThemeInterface.ts:11]: @acro5piano yes, let's change that name for `even` and `uneven` 👍

Comment (titanve) [reply src/theme/ThemeInterface.ts:11]: @acro5piano I will leave the names like this: `even` and `odd`, what do you think?

**Comment by titanve** [comment]
@acro5piano How can I test these changes in my app? I'm trying this without success:
`yarn add https://github.com/acro5piano/react-native-big-calendar.git\#titanve:add-grid-colors`

**Comment by acro5piano** [comment]
> @acro5piano How can I test these changes in my app? I'm trying this without success: `yarn add https://github.com/acro5piano/react-native-big-calendar.git\#titanve:add-grid-colors`

Are you able to run `yarn storybook`? Then you can create a test case with isolated environment. Thanks!

**Comment by titanve** [comment]
> 

@acro5piano where should I run `yarn storybook`?

In my local app directory? or, in the forked repo from `react-native-big-calendar`?

**Comment by acro5piano** [comment]
> In the forked repo it does not work:

Please run `yarn install` before anything

**Comment by titanve** [comment]
> > In the forked repo it does not work:
> 
> Please run `yarn install` before anything

@acro5piano I'm not seeing anything:
image
image

File: src/theme/defaultTheme.ts [hunk 3]
```diff
@@ -6,6 +6,8 @@ export const defaultTheme: ThemeInterface = {
     primary: {
       main: 'rgb(66, 133, 244)',
       contrastText: '#fff',
+      evenCellBg: '#fff',
+      oddCellBg: '#fff',
     },
     nowIndicator: 'red',
```
Comment (acro5piano) [inline src/theme/defaultTheme.ts:12]:
```suggestion
    },
    evenCellBg: '#fff',
    oddCellBg: '#fff',
    nowIndicator: 'red',
```

File: src/theme/defaultTheme.ts [hunk 4]
```diff
@@ -22,9 +24,9 @@ export const defaultTheme: ThemeInterface = {
     },
   },
   eventCellOverlappings: [
-    { main: '#E26245', contrastText: '#fff' }, // orange
-    { main: '#4AC001', contrastText: '#fff' }, // green
-    { main: '#5934C7', contrastText: '#fff' }, // purple
+    { main: '#E26245', contrastText: '#fff', evenCellBg: '#fff', oddCellBg: '#fff' }, // orange
+    { main: '#4AC001', contrastText: '#fff', evenCellBg: '#fff', oddCellBg: '#fff' }, // green
+    { main: '#5934C7', contrastText: '#fff', evenCellBg: '#fff', oddCellBg: '#fff' }, // purple
```
Comment (acro5piano) [inline src/theme/defaultTheme.ts:29]: revert

File: stories/desktop.stories.tsx [hunk 5]
```diff
@@ -286,6 +286,8 @@ storiesOf('showcase - Desktop', module)
               primary: {
                 main: 'purple',
                 contrastText: '#fff',
+                evenCellBg: '#fff',
+                oddCellBg: '#fff',
```
Comment (acro5piano) [inline stories/desktop.stories.tsx:290]: move

File: stories/themes.ts [hunk 6]
```diff
@@ -7,6 +7,8 @@ export const themes: Record<string, PartialTheme> = {
       primary: {
         main: '#6185d0',
         contrastText: '#000',
+        evenCellBg: '#000',
+        oddCellBg: '#000',
```
Comment (acro5piano) [inline stories/themes.ts:11]: move

File: stories/themes.ts [hunk 7]
```diff
@@ -22,16 +24,46 @@ export const themes: Record<string, PartialTheme> = {
...
+        evenCellBg: '#f1f1f1',
+        oddCellBg: '#fff',
+      },
+    },
+    eventCellOverlappings: [
+      {
+        main: '#17651a',
+        contrastText: '#fff',
+        evenCellBg: '#fff',
+        oddCellBg: '#fff',
```
Comment (acro5piano) [inline stories/themes.ts:60]: remove

File: src/components/HourGuideCell.tsx [hunk 8]
```diff
@@ -4,17 +4,22 @@ import { TouchableWithoutFeedback, View } from 'react-native'
...
   hour: number
+  index: number
 }
 
-export const HourGuideCell = ({ cellHeight, onPress, date, hour }: HourGuideCellProps) => {
+export const HourGuideCell = ({ cellHeight, onPress, date, hour, index }: HourGuideCellProps) => {
   const theme = useTheme()
 
+  const evenCellBg = theme.palette.primary.evenCellBg
+  const oddCellBg = theme.palette.primary.oddCellBg
```
Comment (acro5piano) [inline src/components/HourGuideCell.tsx:21]:
```suggestion
  const evenCellBg = theme.palette.evenCellBg
  const oddCellBg = theme.palette.oddCellBg
```

**Review by acro5piano — CHANGES_REQUESTED** [review]
Sorry for my slow review. Nice feature, I like it!

Can you move the new theme property into an upper level as shown in the review comments? Then I'll merge it as soon as possible.

**Comment by titanve** [comment]
@acro5piano can you review again, please?

**Review by acro5piano — APPROVED** [review]

**Comment by acro5piano** [comment]
Released on v2.5.0
Thanks for your contribution!

**Pull request merged**